xline-client/src/lib.rs

```diff
@@ -204,13 +207,32 @@ impl Client {
     ///
     /// If `Self::build_channel` fails.
     #[inline]
-    pub async fn connect(
-        all_members: HashMap<String, String>,
-        options: ClientOptions,
-    ) -> Result<Self> {
+    #[allow(clippy::pattern_type_mismatch)] // allow mismatch in map
+    pub async fn connect<E, S>(all_members: S, options: ClientOptions) -> Result<Self>
+    where
+        E: AsRef<str>,
+        S: IntoIterator<Item = (E, E)> + Clone,
```

Comment:
We can remove the Clone trait bound here, so that users can pass in a wider range of possible types.

Reply:
done